### PR TITLE
Rover: add arming check for windvane if sailing enabled

### PR DESCRIPTION
**Pull request by IamPete1**
Sailing won't work with no windvane, we should be warning of this

### CODE REVIEW

**Comment by Hwurzburg** [comment]
So, this statement in the wiki is wrong? "Although not absolutely required for best results a wind vane should be fitted to allow the vehicle to sense the wind direction."

**Comment by IamPete1** [comment]
@Hwurzburg We don't need a physical windvne, but we do need it to be enabled, the first type is a 'home heading' that just assumes the true wind is coming from the direction pointed when arming.

**Comment by Hwurzburg** [comment]
Dont think that this is very clear in the wiki...probably should be specifically addressed in the Configuration section? maybe with cross link to https://ardupilot.org/rover/docs/wind-vane-home-heading.html ?
I can add if you wish

**Comment by IamPete1** [comment]
@Hwurzburg That would be great, Thank you!

File: Rover/AP_Arming.cpp [hunk 1]
```diff
@@ -82,6 +82,11 @@ bool AP_Arming_Rover::pre_arm_checks(bool report)
         return false;
     }
 
+    if (rover.g2.sailboat.sail_enabled() & !rover.g2.windvane.enabled()) {
```
Comment (khancyr) [inline Rover/AP_Arming.cpp:85]: && ?
maybe create a boat_checks could be good

Comment (IamPete1) [reply Rover/AP_Arming.cpp]: fixed, also in the existing function ;) , Sailboats don't actuality have to be boats (despite the name). Happy to put this in a different function, seems abit much for just this one new check tho.

**Comment by rmackay9** [comment]
Merged, thanks!

A bit shocking that the existing issue with "&" was in there.

File: Rover/AP_Arming.cpp (hunk 1 above)
Comment (peterbarker) [reply Rover/AP_Arming.cpp]: These were supposed to be `&`, not `&&` - that way we run all of the checks rather than failing at the first one!

Comment (rmackay9) [reply Rover/AP_Arming.cpp]: @IamPete1, how'd you like to make a small PR to revert the 2nd change?  I think we should add a comment as well explaining why it's single ampersand.

**Pull request merged**